### PR TITLE
Set toolchains properly in smoke tests

### PR DESCRIPTION
**Pull request by jbartok**

### CODE REVIEW

**Comment by jbartok** [comment]
@bot-gradle test QFL

**Comment by jbartok** [comment]
@bot-gradle test RFM

**Review by wolfs — APPROVED** [review]
LGTM!

File: subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AbstractSmokeTest.groovy [hunk 1]
```diff
@@ -262,6 +263,9 @@ abstract class AbstractSmokeTest extends Specification {
             .withTestKitDir(IntegrationTestBuildContext.INSTANCE.gradleUserHomeDir)
             .withProjectDir(testProjectDir)
             .forwardOutput()
+            .withArguments("-Porg.gradle.java.installations.paths=${AvailableJavaHomes.getAvailableJvms().collect { it.javaHome.absolutePath }.join(",")}")
+            .withArguments("-Porg.gradle.java.installations.auto-detect=false")
+            .withArguments("-Porg.gradle.java.installations.auto-download=false")
```
Comment (wolfs) [inline subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AbstractSmokeTest.groovy:268]: The test is still flaky: https://ge.gradle.org/s/glphel6qrwlfe/tests/:smoke-test:gradleBuildSmokeTest/org.gradle.smoketests.GradleBuildSmokeTest/can%20build%20Gradle%20distribution?top-execution=1

I think the problem is that `withArguments` sets all arguments, so you'd need to add this list similarly to the other arguments below.

**Review by wolfs — CHANGES_REQUESTED** [review]

**Comment by jbartok** [comment]
@bot-gradle test QFL

File: subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AbstractSmokeTest.groovy (hunk 1 above)
Comment (jbartok) [reply subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AbstractSmokeTest.groovy]: Made the change.

File: subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AbstractSmokeTest.groovy [hunk 2]
```diff
@@ -263,7 +264,12 @@ abstract class AbstractSmokeTest extends Specification {
             .withProjectDir(testProjectDir)
             .forwardOutput()
             .withArguments(
-                tasks.toList() + outputParameters() + repoMirrorParameters() + configurationCacheParameters()
+                tasks.toList() + outputParameters() + repoMirrorParameters() + configurationCacheParameters() +
+                [
```
Comment (wolfs) [inline subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AbstractSmokeTest.groovy:268]: Can you extract this to its own method, so it is similar to the other parameter sets?

Comment (wolfs) [inline subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AbstractSmokeTest.groovy:268]: Looks like you need to make sure this is actually a list of String: https://ge.gradle.org/s/cajgqqth6vk34/tests/:smoke-test:gradleBuildSmokeTest/org.gradle.smoketests.GradleBuildBinaryDistributionConfigurationCacheSmokeTest/can%20build%20and%20install%20Gradle%20binary%20distribution%20with%20configuration%20cache%20enabled?focused-exception-line=0-0&top-execution=1

`GString` doesn't work out of the box. Check the other methods.

**Review by wolfs — CHANGES_REQUESTED** [review]

**Comment by jbartok** [comment]
@bot-gradle test QFL

**Review by wolfs — APPROVED** [review]
Awesome, now this seems to work! Thank you!

**Comment by jbartok** [comment]
@bot-gradle test and merge

**Pull request merged**